### ISSUE TITLE
CORS error in embedded video after upgrade from 6.1 to 6.3.1

### ISSUE THREAD

**Issue by trololo9000** [Chocobozzz/PeerTube#6638]
### Describe the problem

When using on nginx, no Access-Control-Allow-Origin header is set for requests in /client/... , causing embedded videos (<iframe src="https://_[mysite]_/videos/embed/_[videoid]_?peertubeLink=0&p2p=0&loop=0&autoplay=0&title=0&muted=0&controls=1" sandbox="allow-scripts allow-popus" ></iframe>) to show up as a strange password form instead of a video. I could fix it for me by adding

> add_header Access-Control-Allow-Origin  '*';

to the

> location ~ ^/client/(.*\.(js|css|png|svg|woff2|otf|ttf|woff|eot))$ {

block in /etc/nginx/sites-available/peertube. At first I thought I might have missed some changes from the support/peertube/nginx/peertube reference version but line-by-line comparison shows that the file seems to be ok and up to date.

### Additional information

* PeerTube instance URL:
* PeerTube version: 6.3.1
* NodeJS version: 20.5.1
* Ffmpeg version: 5.1.6-0+deb12u1
* Redis version: 7.0.15
* PostgreSQL version: 15.8

**Comment by Chocobozzz** [comment]
Hi,

Can you paste the URL that embeds peertube videos so I can reproduce (please remove your workaround so I can see what happens :) )

**Comment by trololo9000** [comment]
I can't because I use it to host product presentation videos in production. :-( But I was able replicate this with other random 6.3.1 peertube+nginx intances in the internet. I also noticed that the embed-code generated by the peertube share dialog adds "allow-same-origin" to the sandbox attribute of the iframe, which makes it work, while my iframe code didn't, which makes it fail when the Access-Control-Allow-Origin header isn't set.
Example: https://www.toetensen-project.com/peertubetest.html

**Comment by Chocobozzz** [comment]
Thanks. Yes PeerTube requires this embed attribute to work. You need to use the embed code provided by peertube. Do you know why you did not set it the first time?

**Comment by trololo9000** [comment]
I don't remember, sorry. I think the code was build more by reading the API documentation then on an actual iframe embed code. Also there is a general recommendation that same-origin and allow scripts shoudn't be used at the same time when there is no urgent necessity. (Because then the iframe can disable the sandbox when it has the same origin as the outer container, which in my case it never has)

**Comment by vincib** [comment]
Hi. I have the same issue in the backoffice of a Wordpress when I embed a PT video inside it.

The embed has been copied / pasted from PT webpage like this 

```
<iframe title="Le rôle des médiateurs à Vandœuvre" width="560" height="315" src="https://webtv.vandoeuvre.net/videos/embed/wCUV9VjGBJsgetRU3544NS" allow="fullscreen" sandbox="allow-same-origin allow-scripts allow-popups allow-forms" style="border: 0px;"></iframe>
```

but I see the following messages in the console : 

Access to CSS stylesheet at 'https://webtv.vandoeuvre.net/client/standalone/videos/assets/embed-BWdoC_q4.css' from origin 'null' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.


and while writing this I found the reason : 

in the wordpress editor, when you do a "custom HTML" element, it embeds it in an iframe with a srcdoc="some html" 
and the srcdoc contains, amongst other, ... your iframe

and in this iframe-in-iframe situation, wordpress does NOT add **allow-same-origin** to the PARENT iframe ...

So this is in fact a wordpress bug...

just a comment there then ...
